scripts/mbedtls_framework/test_case.py

```diff
@@ -89,3 +92,35 @@ def write_data_file(filename: str,
             tc.write(out)
         out.write('\n# End of automatically generated file.\n')
     os.replace(tempfile, filename)
+
+def legacy_domain_hash_dependency_symbol(psa_hash_alg: str) -> str:
+    """Get the dependency symbol for an hash algorithm when the algorithm
+       support is needed by legacy domain code.
+
+       psa_hash_alg  PSA macro of an hash algorithm (e.g. PSA_ALG_SHA_256)
+
+       If invoked in the context of Mbed TLS 4.x or TF-PSA-Crypto, the function
+       just returns the PSA_WANT_ prefixed symbol that corresponds to
+       psa_hash_alg.
+
+       If invoked in the context of Mbed TLS 3.6, the function returns the
+       MBEDTLS_MD_CAN_ prefixed symbol that corresponds to psa_hash_alg.
+
+       The function should be used only for dependencies of legacy domain code.
+       Otherwise, just use the PSA_WANT_ prefixed symbols. For more information
+       about the legacy domain and MBEDTLS_MD_CAN_ prefixed symbols, see
+       transition-guards.md.
+    """
+    info = psa_information.Information()
```

Comment:
Gathering information is costly (parsing several files) and should be cached.

But I'm not convinced this actually needs any information about PSA, as opposed to ad hoc text-based rules. We have a fixed set of algorithms that require adptation for 3.6.

Reply:
Yes that's possible, we could just have a dictionary with PSA_ALG_SHA_... as keys and MBEDTLS_MD_CAN_ ... as values. I am fine with that.